src/viam/ur/module/ur_arm_state_controlled.cpp

```diff
@@ -62,56 +62,111 @@ std::optional<URArm::state_::event_variant_> URArm::state_::state_controlled_::h
+                            urcl::control::TrajectoryControlMessage::TRAJECTORY_CANCEL, 0, RobotReceiveTimeout::off())) {
+                        mr.cancel_error("failed to send trajectory cancel");
+                        VIAM_SDK_LOG(error) << "cancel failed; dropping connection";
+                        return event_connection_lost_::trajectory_control_failure();
+                    }
+                    return std::nullopt;
+                } else if (!cmd.empty() && mr.cancellation_request) {
+                    // We have a move request that we haven't issued but a
+                    // cancel is already pending. Don't issue it, just cancel it.
+                    std::exchange(state.move_request_, {})->complete_cancelled();
+                    return std::nullopt;
+
+                } else {
+                    // TODO: is it assured that we have positions/velocities here?
+                    mr.write_joint_data(state.ephemeral_->joint_positions, state.ephemeral_->joint_velocities);
+                    return std::nullopt;
+                }
+
+            } else if constexpr (std::is_same_v<T, std::optional<pose_sample>>) {
+                // Operating in pose-space
+
+                if (cmd.has_value() && !mr.cancellation_request) {
+                    // Send the single pose command
+                    if (!arm_conn_->driver->writeTrajectoryControlMessage(
+                            urcl::control::TrajectoryControlMessage::TRAJECTORY_START, 1, RobotReceiveTimeout::off())) {
+                        VIAM_SDK_LOG(error) << "single pose: start failed; dropping connection";
+                        std::exchange(state.move_request_, {})->complete_error("failed to send trajectory start for single pose");
+                        return event_connection_lost_::trajectory_control_failure();
+                    }
+
+                    auto ps = std::move(*cmd);
+                    VIAM_SDK_LOG(info) << "URArm sending single pose (tool-space)";
+
+                    // TODO(RSDK-12294): determine how to set the velocity and acceleration so they are not hardcoded
+                    const float velocity = 0.25F;
+                    const float acceleration = 0.5F;
+                    const float timestep = 0;
+                    const float blend_radius = 0;
+                    const bool cartesian = true;
+
+                    if (!arm_conn_->driver->writeTrajectoryPoint(ps.p, acceleration, velocity, cartesian, timestep, blend_radius)) {
+                        VIAM_SDK_LOG(error) << "single pose: writeTrajectoryPoint failed; dropping connection";
+                        std::exchange(state.move_request_, {})->complete_error("failed to send single pose point");
+                        return event_connection_lost_::trajectory_control_failure();
+                    }
+
+                    // Mark as sent by setting to nullopt
+                    cmd = std::nullopt;
```

Comment:
I think this should work with `emplace`, not sure which is cleaner?

Reply:
Or, maybe even better, just `.reset()`